### ISSUE TITLE
Rename `max_num_nodes_per_graph` to `batch_num_nodes_per_graph`

### ISSUE THREAD

**Issue by DomInvivo** [datamol-io/graphium#237]
The `max_num_nodes_per_graph` is confusing, as it can refer to the one used in the batch size such that `max_num_nodes_per_graph * batch_size == max_nodes_in_a_batch`. This is different from the other `max_num_nodes_per_graph` that is meant for the actual larger graph in the dataset. This is super confusing.

Rename the right instances to `batch_num_nodes_per_graph`

**Comment by WenkelF** [comment]
@DomInvivo I will fix this in collate.py.

Any other files to take a look at for this?

**Comment by DomInvivo** [comment]
@WenkelF, don't fix that in the `collate.py`. I think the naming is right there. The confusion comes from `max_num_nodes_per_graph` sometimes meaning the actual maximum number of nodes for all graphs (such as in `collate.py`), but other time it means the maximum number of nodes allowed in a batch (such as in the `ipu` folder)

I would say to change these places:
- All instances in the `.yaml` config files, since they refer to the number of nodes per graph in a batch when using padding, such as on IPU.
- All instances of every python file in the `ipu` folder
- All instances in the `nn` folder
- Basically, only those in the `data` folder remain untouched

Don't forget to change the tests accordingly

**Comment by DomInvivo** [comment]
As discussed, let's change the `collate.py` as well :)

**Comment by DomInvivo** [comment]
Closing this issue as it doesn't seem like a good idea in retrospective

**Comment by DomInvivo** [comment]
Closing since it causes more confusion to rename it
